File: app/src/main/java/com/example/oldbooks/MainActivity.java
package com.example.oldbooks;

import androidx.appcompat.app.AppCompatActivity;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Toast;

import com.example.customadspackage.GoogleAdMobManager;
import com.example.oldbooks.databinding.ActivityMainBinding;
import com.example.oldbooks.extras.Preconditions;
import com.example.oldbooks.manager.CoinManager;
import com.example.oldbooks.manager.FirebaseManager;
import com.example.oldbooks.manager.UserManager;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding act_binding;
    private Activity activity;
    private final String TAG = "MainActivity";
    private FirebaseManager firebaseManager;
    Runnable addCoinsCallback;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        act_binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(act_binding.getRoot());
        activity = this;

        GoogleAdMobManager.getInstance().Initialize(activity);
        AppController.getInstance();
        firebaseManager = AppController.getInstance().getManager(FirebaseManager.class);

        Intent intent = getIntent();
        if (intent != null && intent.hasExtra("user")) {
            User user = (User) intent.getSerializableExtra("user");
        }

        act_binding.btnAddPost.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Log.d(TAG, "AddPost Clicked");
                startActivity(new Intent(activity,PublishPost.class));
            }
        });
        act_binding.btnViewPost.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Log.d(TAG, "AddPost Clicked");
                startActivity(new Intent(activity, PostList.class));
            }
        });
        act_binding.btnShowRewardAd.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Log.d(TAG, "btnShowRewardAd clicked");
                GoogleAdMobManager.getInstance().ShowRewardedAd(MainActivity.this, addCoinsCallback);
            }
        });
        act_binding.btnShowRewardInterAd.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Log.d(TAG, "btnShowRewardInterAd clicked");
                GoogleAdMobManager.getInstance().ShowRewardedInterstitialAd(MainActivity.this, addCoinsCallback);
            }
        });
        act_binding.btnDailyReward.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Log.d(TAG, "btnDailyReward clicked");
                startActivity(new Intent(activity, DailyRewards.class));
            }
        });
        addCoinsCallback = new Runnable() {
            @Override
            public void run() {
                AppController.getInstance().getManager(CoinManager.class).addCoins(10);
                Log.d(TAG, "give addCoins(10)");
                Toast.makeText(activity, "Reward Given: +10 coins", Toast.LENGTH_SHORT).show();
            }
        };
        act_binding.btnSignup.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                btnSignUp();
            }
        });
        act_binding.btnLogin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                btnLogin();
            }
        });
    }

// endregion

// region Login/Signup

//region Signup
    public void btnSignUp(){
        User user = new User();
        user.setUsername(act_binding.edtUserName.getText().toString());
        user.setPassword(act_binding.edtSignupPswd.getText().toString());
        user.setPhoneNumber(act_binding.edtPhoneNumber.getText().toString());
        if (!user.validateUsername(user.getUsername()) && !user.validatePassword(user.getPassword()) && !user.validatePhoneNumber(user.getPhoneNumber()))
        {
            return;
        }
//        if (!verifySignup(user))
//        {
//            return;
//        }
        user.setJoinedDate(AppController.getCurrentTimestamp());
        firebaseManager.addNewUser(activity, user);
    }
    public boolean verifySignup(User user){
        return firebaseManager.verifySignupCredential(activity, user.getUsername());
    }
// endregion Signup

//region Login
    public void btnLogin(){
        User user = new User();
        user.setUsername(act_binding.edtId.getText().toString());
        user.setPassword(act_binding.edtId.getText().toString());
        if (!user.validateUsername(user.getUsername()) && !user.validatePassword(user.getPassword()))
        {
            return;
        }
//        if (!verifyLogin(user))
//        {
//            return;
//        }
        user.setUserStatus(Enums.UserStatus.ACTIVE);
        firebaseManager.loginUser(activity,user);
    }
    public boolean verifyLogin(User user){
        return firebaseManager.verifyLoginCredential(activity, user.getUsername());
    }
// endregion Login


// endregion Login/Signup
}

File: app/src/main/java/com/example/oldbooks/MainActivity.kt
package com.example.oldbooks

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.Toast
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import com.example.oldbooks.databinding.ActivityLoginSignupBinding
import com.example.oldbooks.ui.theme.OldBooksTheme

class MainActivity : ComponentActivity() {
    private lateinit var binding: ActivityLoginSignupBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityLoginSignupBinding.inflate(layoutInflater)
        val view = binding.root
        setContent {
            OldBooksTheme {
                // A surface container using the 'background' color from the theme
                Surface(
                    modifier = Modifier.fillMaxSize(),
                    color = MaterialTheme.colorScheme.background
                ) {
                    Greeting("Android")
                }
            }
        }
        setContentView(view)
        binding.addPost.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View) {
                // Add the functionality you want when the button is clicked
                // For example, display a toast message
                Toast.makeText(applicationContext, "Button AddPost Clicked!", Toast.LENGTH_SHORT).show()
                // Create an intent to switch to the new activity
                val intent = Intent(applicationContext, PublishPost::class.java)
                startActivity(intent)
            }
        })
        binding.viewPost.setOnClickListener(object : View.OnClickListener{
            override fun onClick(p0: View?) {
                Toast.makeText(applicationContext, "Button ViewPost Clicked!", Toast.LENGTH_SHORT).show()
                val intent = Intent(applicationContext, PublishPost::class.java)
                startActivity(intent)
            }
        })
    }
}

@Composable
fun Greeting(name: String, modifier: Modifier = Modifier) {
    Text(
        text = "Hello $name!",
        modifier = modifier
    )
}

@Preview(showBackground = true)
@Composable
fun GreetingPreview() {
    OldBooksTheme {
        Greeting("Android")
    }
}